src/sentry/issues/related/same_root_cause.py

```diff
@@ -7,7 +7,7 @@
 from sentry.utils.query import RangeQuerySetWrapper
 
 
-def same_root_cause_analysis(group: Group) -> list[int]:
+def same_root_cause_analysis(group: Group) -> tuple[list[int], dict[str, str]]:
```

Comment:
I personally like `dict` returns since they have to name each return value, but either way I think we should extract this to a type alias and reuse it for both algos, maybe even typing `RELATED_ISSUES_ALGORITHMS` explicitly so that new entries to that dict have a blueprint.